plugin/pulpcore/plugin/download/asyncio/factory.py

```diff
@@ -0,0 +1,116 @@
+import aiohttp
+import atexit
+from gettext import gettext as _
+import ssl
+from urllib.parse import urlparse
+
+from .http import HttpDownloader
+
+
+class DownloadFactory:
+    """
+    A factory for creating configured downloader objects that all share a single session.
+
+    Usage:
+        >>> import asyncio
+        >>> loop = asyncio._get_running_loop()
```

Comment:
Did you mean `_get_running_loop()`?  It's named to be non-public.

Reply:
yeah this was wrong. I updated it with the next push.